src/dtls.c

```diff
@@ -290,60 +343,385 @@ static int TlsSessionIdIsValid(WOLFSSL* ssl, WolfSSL_ConstVector sessionID,
 #endif
     ret = TlsSessionCacheGetAndLock(sessionID.elements, &sess, &sessRow);
     if (ret == 0 && sess != NULL) {
-        *isValid = 1;
+        /* Store info for later */
+        pskInfo->pv = sess->version;
+        pskInfo->cipherSuite0 = sess->cipherSuite0;
+        pskInfo->cipherSuite = sess->cipherSuite;
+        pskInfo->namedGroup = sess->namedGroup;
+
+        pskInfo->isValid = 1;
         TlsSessionCacheUnlockRow(sessRow);
     }
 
     return 0;
 }
 
-static int TlsResumptionIsValid(WOLFSSL* ssl, WolfSSL_CH* ch, byte* isValid)
+static int TlsResumptionIsValid(const WOLFSSL* ssl, WolfSSL_CH* ch,
+                                PskInfo* pskInfo)
 {
     int ret;
 
-    *isValid = 0;
 #ifdef HAVE_SESSION_TICKET
-    ret = TlsTicketIsValid(ssl, ch->extension, isValid);
+    ret = TlsTicketIsValid(ssl, ch->extension, pskInfo);
     if (ret != 0)
         return ret;
-    if (*isValid)
+    if (pskInfo->isValid)
         return 0;
 #endif /* HAVE_SESSION_TICKET */
-    ret = TlsSessionIdIsValid(ssl, ch->sessionId, isValid);
+    ret = TlsSessionIdIsValid(ssl, ch->sessionId, pskInfo);
     return ret;
 }
 #endif /* WOLFSSL_DTLS_NO_HVR_ON_RESUME */
 
-int DoClientHelloStateless(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
-    word32 helloSz, byte* process)
+#ifdef WOLFSSL_DTLS13
+static int TlsCheckSupportedVersion(const WOLFSSL* ssl,
+        WolfSSL_CH* ch, byte *isTls13, PskInfo* pskInfo)
+{
+    WolfSSL_ConstVector tlsxSupportedVersions;
+    int ret;
+    ProtocolVersion pv = ssl->version;
+
+    ret = TlsxFindByType(&tlsxSupportedVersions, TLSX_SUPPORTED_VERSIONS,
+                         ch->extension);
+    if (ret != 0)
+        return ret;
+    if (tlsxSupportedVersions.size == 0) {
+        *isTls13 = 0;
```

Comment:
What do you mean? Check if the session object has the same version?

Reply:
yes

Reply:
There is already a check right after
```
    if (pskInfo->isValid && (pskInfo->pv.major != pv.major ||
                             pskInfo->pv.minor != pv.minor))
        return VERSION_ERROR;
```

Reply:
I see, I was referring to the scenario below:
- TlsResumptionIsValid is invoked and returns a valid "session" in pskInfo
- There are no VersionSupported extensions (`tlsxSupportedVersion.size == 0` above).
But upon further consideration, it is not necessary to check the version in DTLS version 1.2 or lower because the server does not have to reply with a supported version extension.